airflow/jobs/scheduler_job.py

```diff
@@ -745,6 +757,9 @@ def _execute(self) -> None:
 
             self._run_scheduler_loop()
 
+            if self.enabled_dagrun_listener:
```

Comment:
yeah like ... what do you think about that @mobuchowski ?  if it is feasible to add a null notifier, i.e. some kind of base class that doesn't do anything, then maybe we could get rid of all of these if checks, and maybe that would be better. what do you think of the thought?

Reply:
We can just check for that in `notify_dagrun_state_changed`, and I changed it to just do that